Tests/Get-BMDeployment.Tests.ps1

```diff
@@ -212,9 +212,177 @@ Describe 'Get-BMDeployment' {
         $deployment3 = GivenDeployment $build2
         WhenGettingBMDeployment -Deployment $deployment.id
         ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 1
         ThenDeploymentShouldBeReturned $deployment
         ThenDeploymentShouldNotBeReturned $deployment2
         ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get a deployment by application name' -Skip {

```

Comment:
Why are you skipping all these tests?

Reply:
There's a bug in the Deployments API that impacts this function, but that doesn't mean we should be skipping the tests.

Instead update the function and tests to workaround the bug by including filtering parameters in the API request.

Reply:
Figured out how to work around the bug by checking the fields of the returned objects rather then comparing the Deployment ID's.